Makefile

```diff
@@ -213,6 +213,59 @@ $(NODE_CONTAINER_BIN_DIR)/birdcl:
 	$(CURL) -L $(BIRDCL_URL) -o $(@D)/birdcl
 	chmod +x $(@D)/*
 
+# ensure we have a real imagetag
+imagetag:
+ifndef IMAGETAG
+	$(error IMAGETAG is undefined - run using make <target> IMAGETAG=X.Y.Z)
+endif
+
+## push one arch
+push: imagetag
+	docker push $(NODE_CONTAINER_NAME):$(IMAGETAG)-$(ARCH)
+	docker push quay.io/$(NODE_CONTAINER_NAME):$(IMAGETAG)-$(ARCH)
+ifeq ($(ARCH),amd64)
+	docker push $(NODE_CONTAINER_NAME):$(IMAGETAG)
+	docker push quay.io/$(NODE_CONTAINER_NAME):$(IMAGETAG)
+
+ifeq ($(RELEASE),true)
+	# If this is a release, also push the GKE images to gcr. 
+	docker push gcr.io/projectcalico-org/node:$(IMAGETAG)
+	docker push eu.gcr.io/projectcalico-org/node:$(IMAGETAG)
+	docker push asia.gcr.io/projectcalico-org/node:$(IMAGETAG)
+	docker push us.gcr.io/projectcalico-org/node:$(IMAGETAG)
+endif
+endif
+
+push-all: imagetag $(addprefix sub-push-,$(ARCHES))
+sub-push-%:
+	$(MAKE) push ARCH=$* IMAGETAG=$(IMAGETAG)
+
+## tag images of one arch
+tag-images: imagetag
+	docker tag $(NODE_CONTAINER_NAME):latest-$(ARCH) $(NODE_CONTAINER_NAME):$(IMAGETAG)-$(ARCH)
```

Comment:
Feels like this might be getting out of control.  Could we have a variable that contains the list of repos to push to:
```
DOCKER_REMOTES:=quay.io/ 
ifeq ($(RELEASE),true)
DOCKER_REMOTES+=gcr.io/ eu.gcr.io/ ...
endif
```
and then loop over it in `tag-images` and `push`?

Reply:
SGTM - one thing that makes this a bit trickier is that the org is different across different repos.

e.g. calico vs projectcalico-org.

I've done that by putting that in the list as well, but it means the NODE_CONTAINER_NAME override won't work for the re-tagged images. I think that is OK - but LMK if there's a reason that doesn't work.